core/src/test/java/org/apache/iceberg/TestBaseIncrementalAppendScan.java

```diff
@@ -101,6 +226,39 @@ public void testToSnapshot() {
     Assert.assertEquals(2, Iterables.size(scan.planFiles()));
   }
 
+  @Test
+  public void testToSnapshotWithRef() {
+    table.newFastAppend().appendFile(FILE_A).commit();
+    long snapshotAId = table.currentSnapshot().snapshotId();
+    table.newFastAppend().appendFile(FILE_B).commit();
+    long snapshotBId = table.currentSnapshot().snapshotId();
+
+    String branchName = "b1";
+    table.manageSnapshots().createBranch(branchName, snapshotBId).commit();
+
+    String tagSnapshotMainBName = "t1";
+    table.manageSnapshots().createTag(tagSnapshotMainBName, snapshotBId).commit();
+
+    String tagSnapshotBranchBName = "t2";
+    table.newFastAppend().appendFile(FILE_C).toBranch(branchName).commit();
+    long snapshotBranchBId = table.snapshot(branchName).snapshotId();
+    table.manageSnapshots().createTag(tagSnapshotBranchBName, snapshotBranchBId).commit();
+
+    IncrementalAppendScan scan = newScan().toSnapshot(tagSnapshotMainBName);
+    Assertions.assertThat(scan.planFiles()).hasSize(2);
+
+    IncrementalAppendScan scan2 = newScan().toSnapshot(tagSnapshotBranchBName);
+    Assertions.assertThat(scan2.planFiles()).hasSize(3);
+
+    Assertions.assertThatThrownBy(() -> newScan().toSnapshot(branchName))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage(String.format("Ref %s is not a tag", branchName));
+
+    Assertions.assertThatThrownBy(() -> newScan().toSnapshot("notExistTag"))
```

Comment:
same as above w.r.t splitting out success/failure testing and separating tests against branches/tags